src/main/java/org/fenixedu/academic/util/AcademicPhoneNumberHandler.java

```diff
@@ -0,0 +1,112 @@
+package org.fenixedu.academic.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.fenixedu.academic.domain.contacts.EmailAddress;
+import org.fenixedu.academic.domain.contacts.MobilePhone;
+import org.fenixedu.academic.domain.contacts.PartyContact;
+import org.fenixedu.academic.domain.contacts.Phone;
+import org.fenixedu.academic.domain.contacts.PhysicalAddress;
+import org.fenixedu.academic.domain.contacts.WebAddress;
+import org.fenixedu.academic.util.PhoneUtil.PhoneNumberHandler;
+
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.PhoneNumberUtil.PhoneNumberFormat;
+import com.google.i18n.phonenumbers.PhoneNumberUtil.PhoneNumberType;
+import com.google.i18n.phonenumbers.Phonenumber.PhoneNumber;
+
+public class AcademicPhoneNumberHandler implements PhoneNumberHandler {
+
+    protected static final PhoneNumberUtil PHONE_UTIL = PhoneNumberUtil.getInstance();
+
+    private static final Map<Class<? extends PartyContact>, ContactResolver> DEFAULT_RESOLVERS = new HashMap<>();
```

Comment:
How can another module add or replace a ContactResolver?
Don't we need an extra method in this class to register a ContactResolver?